### PR TITLE
[CodeQuality][DeadCode] Add For_, Closure, and Finally_, Case_, Catch_ to NodeTypeGroup::STMTS_AWARE

### PR DESCRIPTION
**Pull request by samsonasik**
- Add For_, Closure, and Finally_, Case_, Catch_ to NodeTypeGroup::STMTS_AWARE
- re-use in RemoveUnreachableStatementRector
- apply `@param` values on its usage

### CODE REVIEW

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

File: rules/CodeQuality/Rector/ClassMethod/InlineArrayReturnAssignRector.php [hunk 1]
```diff
@@ -69,7 +81,7 @@ public function getNodeTypes(): array
     }
 
     /**
-     * @param ClassMethod|Function_ $node
+     * @param ClassMethod|Function_|If_|Else_|ElseIf_|Do_|Foreach_|TryCatch|While_|For_|Closure|Finally_|Case_|Catch_ $node
```
Comment (samsonasik) [inline rules/CodeQuality/Rector/ClassMethod/InlineArrayReturnAssignRector.php]: @ondrejmirtes is there a way to set instanceof for `@param` which read constant, eg:

```php
     * @param instance-of<NodeTypeGroup::STMTS_AWARE> $node
 ```

`value-of` is for got constant return string of class-string, which is different thing.

Comment (samsonasik) [reply rules/CodeQuality/Rector/ClassMethod/InlineArrayReturnAssignRector.php]: @TomasVotruba another solution is probably use : 

```php
Assert::propertyExists($node, 'stmts');
```

what do you think?

Comment (TomasVotruba) [reply rules/CodeQuality/Rector/ClassMethod/InlineArrayReturnAssignRector.php]: > Is there a way to set instanceof for @param which read constant, eg:

:+1:  That would be great. I was thinking of it, but could not hack around something useful.

> another solution is probably use :

We could try that.

Comment (samsonasik) [reply rules/CodeQuality/Rector/ClassMethod/InlineArrayReturnAssignRector.php]: implemented with `Assert::propertyExists()` https://github.com/rectorphp/rector-src/pull/2258/commits/88d5c3398fcc688e3be80e5b79ff2797239ba983

**Pull request merged**